activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -106,6 +106,14 @@ def self.build_read_query_regexp(*parts) # :nodoc:
         Regexp.union(*parts)
       end
 
+      def self.quoted_column_names
+        @quoted_column_names ||= {}
+      end
+
+      def self.quoted_table_names
```

Comment:
Is it intended to expose these in API doc?

Reply:
Good point, no it's not. I think they should be private. Not sure what the best way to do it.

I can certainly `# nodoc` them, but they are still accessible. I'm open to suggestions.

Reply:
I guess I could use class variables (`@@`) instead, but then I'm afraid about them being inherited, hence shared between distinct adapter implementations, in unexpected ways.

Reply:
Maybe you could make it a protected class method?

Reply:
Nope:

```ruby
class Foo
  class << self
    protected

    def cache
    end
  end

  def access_cache
    self.class.cache
  end
end

Foo.new.access_cache # protected method `cache' called for Foo:Class (NoMethodError)
```